src/main/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -306,7 +324,7 @@ public String toString() {
      * @param tagsToMatch list of tags that must be equal if present in both sequence. Must contain MD, and LN
      * @return dictionary consisting of the same sequences as the two inputs with the merged values of tags.
      */
-    static public SAMSequenceDictionary mergeDictionaries(final SAMSequenceDictionary dict1,
+    public static SAMSequenceDictionary mergeDictionaries(final SAMSequenceDictionary dict1,
```

Comment:
Hmn, do we have to worry about alias name collisions when merging dicionaries?

Reply:
yes. :-)

Reply:
as written, it will only merge dictionaries if all the tags agree (when present). this (implicitly) includes the AN tag, so the two dictionaries will have to have the same alternative names. So as written it is safe. 

If we want the method to merge the following contigs:

dict1: 
@SN ID=name1 AN=n1

and dict2:
@SN ID=n1 AN=name1

or even with dict3:
@SN ID=nana1 AN=name1

then some additional smarts would have to be written....

Reply:
That's probably a good idea provided the md5's are the same, but it can be a different pr...